src/snowflake/cli/plugins/object/stage/manager.py

```diff
@@ -137,3 +147,97 @@ def create(self, stage_name: str, comment: Optional[str] = None) -> SnowflakeCur
         if comment:
             query += f" comment='{comment}'"
         return self._execute_query(query)
+
+    def execute(
+        self,
+        stage_path: str,
+        on_error: OnErrorType,
+        parameters: Optional[List[str]] = None,
+    ):
+        files_list = self._get_files_list_from_stage(stage_path)
+        filtered_list = self._filter_files_list(stage_path, files_list)
+        sorted_list = sorted(
+            filtered_list, key=cmp_to_key(self._stage_files_comparator)
+        )
+
+        sql_parameters = self._parse_execute_parameters(parameters)
+        results = []
+        for file in sorted_list:
+            results.append(self._call_execute_immediate(file, sql_parameters, on_error))
+
+        return results
+
+    def _get_files_list_from_stage(self, stage_path: str) -> List[str]:
+        stage_name = stage_path.split("/")[0]
+        files_list_result = self.list_files(stage_name).fetchall()
```

Comment:
If I'm correct, `list_files` should already work well with path's subdirectories, just the argument name is unfortunate:
```
▶ snow object stage list XD/fromdisc
ls @XD/fromdisc
+---------------------------------------------------------------------------------------------------------+
| name                          | size | md5                              | last_modified                 |
|-------------------------------+------+----------------------------------+-------------------------------|
| xd/fromdisc/file1_on_main.txt | 16   | eaa66e3eff4a7c1f310314fc0d7e44ed | Tue, 27 Feb 2024 16:44:10 GMT |
| xd/fromdisc/file2_on_main.txt | 32   | 2ee58a50ba86be95e57dcaff04a038b3 | Tue, 27 Feb 2024 16:44:09 GMT |
+---------------------------------------------------------------------------------------------------------+
```
We can avoid filtering result and simplify logic